From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Kalyan Alle <kalyan.alle@intel.com>
Date: Mon, 30 Aug 2021 01:24:21 +0800
Subject: [PATCH] Zero copy changes in minigbm-gralloc to fix the android

boot failure issue.

minigbm needs to open virtio-gpu render node and enable the virtiogpu
backend driver to bring up the android boot animation and UI.
---
 Android.bp                          | 10 +++++-----
 Makefile                            |  4 +++-
 cros_gralloc/cros_gralloc_driver.cc |  2 +-
 drv.c                               |  9 +++++++--
 4 files changed, 16 insertions(+), 9 deletions(-)

diff --git a/Android.bp b/Android.bp
index 4aa6a22..b575b13 100644
--- a/Android.bp
+++ b/Android.bp
@@ -39,7 +39,7 @@ cc_defaults {
         "-Wno-switch",
         "-Wno-format",
         "-Wno-unused-variable",
-        "-DDRV_I915",
+        "-DDRV_VIRTIO_GPU",
         "-DUSE_GRALLOC1",
         "-Wno-cast-qual",
     ],
@@ -84,7 +84,7 @@ cc_defaults {
     ],
 
     cflags: [
-        "-DDRV_I915",
+        "-DDRV_VIRTIO_GPU",
         "-DUSE_GRALLOC1",
     ],
 
@@ -138,7 +138,7 @@ cc_library_shared {
             enabled: true,
         },
     },
-    cflags: ["-DDRV_I915"],
+    cflags: ["-DDRV_VIRTIO_GPU"],
     srcs: ["cros_gralloc/gralloc0/gralloc0.cc"],
 }
 
@@ -154,7 +154,7 @@ cc_library_shared {
         },
     },
     cflags: [
-        "-DDRV_I915",
+        "-DDRV_VIRTIO_GPU",
         "-DUSE_GRALLOC1",
     ],
     relative_install_path: "hw",
@@ -173,7 +173,7 @@ cc_library_shared {
         },
     },
     cflags: [
-        "-DDRV_I915",
+        "-DDRV_VIRTIO_GPU",
         "-DUSE_GRALLOC1",
     ],
     relative_install_path: "hw",
diff --git a/Makefile b/Makefile
index 35f92f2..a9b7100 100644
--- a/Makefile
+++ b/Makefile
@@ -34,7 +34,9 @@ endif
 ifdef DRV_VC4
 	CFLAGS += $(shell $(PKG_CONFIG) --cflags libdrm_vc4)
 endif
-
+ifdef DRV_VIRTIO_GPU
+	CFLAGS += $(shell $(PKG_CONFIG) --cflags libdrm_intel)
+endif
 CPPFLAGS += $(PC_CFLAGS)
 LDLIBS += $(PC_LIBS)
 
diff --git a/cros_gralloc/cros_gralloc_driver.cc b/cros_gralloc/cros_gralloc_driver.cc
index fd8d7b3..7808a38 100644
--- a/cros_gralloc/cros_gralloc_driver.cc
+++ b/cros_gralloc/cros_gralloc_driver.cc
@@ -47,7 +47,7 @@ int32_t cros_gralloc_driver::init()
 
 	int fd;
 	drmVersionPtr version;
-	char const *str = "%s/renderD%d";
+	char const *str = "%s/renderD129";
 	const char *undesired[2] = { "vgem", nullptr };
 	uint32_t num_nodes = 63;
 	uint32_t min_node = 128;
diff --git a/drv.c b/drv.c
index c7c487a..3a30063 100644
--- a/drv.c
+++ b/drv.c
@@ -68,8 +68,10 @@ extern const struct backend backend_udl;
 #ifdef DRV_VC4
 extern const struct backend backend_vc4;
 #endif
-extern const struct backend backend_vgem;
+#ifdef DRV_VIRTIO_GPU
 extern const struct backend backend_virtio_gpu;
+#endif
+extern const struct backend backend_vgem;
 
 static const struct backend *drv_get_backend(int fd)
 {
@@ -121,7 +123,10 @@ static const struct backend *drv_get_backend(int fd)
 #ifdef DRV_VC4
 		&backend_vc4,
 #endif
-		&backend_vgem,	    &backend_virtio_gpu,
+#ifdef DRV_VIRTIO_GPU
+		&backend_virtio_gpu,
+#endif
+		&backend_vgem,
 	};
 
 	for (i = 0; i < ARRAY_SIZE(backend_list); i++) {
-- 
2.17.1

